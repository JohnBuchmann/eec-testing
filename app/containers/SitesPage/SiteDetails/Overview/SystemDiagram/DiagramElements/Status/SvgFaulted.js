/**
 * Faulted icon
 */

import React from 'react';
import { Colors } from 'Theme';

/**
 * Display the core of faulted icon for svg
 */
function SvgFaulted() {
  return (
    <>
      <desc>Faulted icon core</desc>
      <defs>
        <filter id="mrqp0xtl3a">
          <feColorMatrix
            in="SourceGraphic"
            values="0 0 0 0 1.000000 0 0 0 0 1.000000 0 0 0 0 1.000000 0 0 0 1.000000 0"
          />
        </filter>
      </defs>
      <g fill="none" fillRule="evenodd">
        <g>
          <g>
            <g>
              <g
                filter="url(#mrqp0xtl3a)"
                transform="translate(-322 -284) translate(98 178) translate(220 102) translate(4 4)"
              >
                <g>
                  <path
                    fill={Colors.black}
                    fillRule="nonzero"
                    d="M16 1.861L14.139 0 8 6.139 1.861 0 0 1.861 6.139 8 0 14.139 1.861 16 8 9.861 14.139 16 16 14.139 9.861 8z"
                    transform="translate(4 4)"
                  />
                </g>
              </g>
            </g>
          </g>
        </g>
      </g>
    </>
  );
}

export default SvgFaulted;
