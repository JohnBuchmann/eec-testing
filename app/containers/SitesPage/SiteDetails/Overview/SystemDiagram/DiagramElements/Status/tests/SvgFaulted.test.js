import React from 'react';
import { shallow } from 'enzyme';
import SvgFaulted from '../SvgFaulted';

describe('SvgFaulted', () => {
  it('should render SVG core structure', () => {
    const actual = shallow(<SvgFaulted />);

    expect(actual).toMatchSnapshot();
  });
});
