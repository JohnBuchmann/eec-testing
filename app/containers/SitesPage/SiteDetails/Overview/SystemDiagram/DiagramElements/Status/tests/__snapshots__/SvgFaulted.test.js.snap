// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SvgFaulted should render SVG core structure 1`] = `
<Fragment>
  <desc>
    Faulted icon core
  </desc>
  <defs>
    <filter
      id="mrqp0xtl3a"
    >
      <feColorMatrix
        in="SourceGraphic"
        values="0 0 0 0 1.000000 0 0 0 0 1.000000 0 0 0 0 1.000000 0 0 0 1.000000 0"
      />
    </filter>
  </defs>
  <g
    fill="none"
    fillRule="evenodd"
  >
    <g>
      <g>
        <g>
          <g
            filter="url(#mrqp0xtl3a)"
            transform="translate(-322 -284) translate(98 178) translate(220 102) translate(4 4)"
          >
            <g>
              <path
                d="M16 1.861L14.139 0 8 6.139 1.861 0 0 1.861 6.139 8 0 14.139 1.861 16 8 9.861 14.139 16 16 14.139 9.861 8z"
                fill="#000000"
                fillRule="nonzero"
                transform="translate(4 4)"
              />
            </g>
          </g>
        </g>
      </g>
    </g>
  </g>
</Fragment>
`;
